src/monattack.cpp

```diff
@@ -4860,3 +4860,78 @@ bool mattack::dodge_check(monster *z, Creature *target){
     }
     return false;
 }
+
+void mattack::ink_jet(monster *z, int index) {
+
+    if (!z->can_act()) {
+        return;
+    }
+
+    int jet_range =  5;
+
+    // TODO: make the creature utilise this ability then escaping the threat
+    Creature *target = z->attack_target();
+
+    if (target == nullptr || rl_dist(z->pos(), target->pos()) > jet_range || !z->sees(*target)) {
+        return;
+    }
+
+    std::vector<tripoint> line = g->m.find_clear_path(z->pos(), target->pos());
+
+    z->reset_special(index);
+
+    z->moves -= 50;
+
+    // Notify the player, if he sees the use of this ability
+    bool u_see = g->u.sees(*z);
+
+    if (u_see || target->is_player()) {
+        const auto msg_type = (target->attitude_to(g->u) == Creature::A_FRIENDLY ? m_bad : m_warning);
+        target->add_msg_player_or_npc(msg_type, (_("The %s squirts a jet of black ink at you!"), z->name().c_str()), 
```

Comment:
Funny syntax here. It doesn't do what you think it does.

A bit shorter, but syntactically the same:
```C++
add_msg_player_or_npc( ( "A", z->name().c_str() ),
                       ( "B", z->name().c_str() ) );
```
This calls `add_msg_player_or_npc` with *two* arguments: the first is `( "A", z->name().c_str() )` and this invokes the comma operator with "A" and `z->name().c_str()` as arguments. The comma operator evaluates is arguments and returns the second (or with multiple commas the last) argument. The code is there equivalent to this:
```C++
add_msg_player_or_npc( z->name().c_str(),
                       z->name().c_str() );
```

You probably want to write this:
```C++
target->add_msg_player_or_npc( msg_type, _("The %s squirts a jet of black ink at you!"),
                                         _("The %s squirts a jet of black ink at <npcname>!"),
                                         z->name().c_str() );
```